src/sonic-yang-mgmt/sonic_yang.py

```diff
@@ -0,0 +1,656 @@
+    returns:  Exception if failed
+    """
+    def merge_data(self, data_file, yang_dir=None):
+        #load all yang models to ctx
+        if not yang_dir:
+            yang_dir = self.yang_dir
+
+        try:
+            ctx = self.load_schema_modules_ctx(yang_dir)
+
+            #source data node
+            source_node = ctx.parse_data_path(str(data_file), ly.LYD_JSON, ly.LYD_OPT_CONFIG | ly.LYD_OPT_STRICT)
+
+            #merge
+            self.root.merge(source_node, 0)
+        except Exception as e:
+            self.fail(e)
+
+    """
+    _delete_node(): delete a node from the schema/data tree, internal function
+    input:    xpath of the schema/data node
+    returns:  True - success   False - failed
+    """
+    def _delete_node(self, xpath=None, node=None):
+        if node is None:
+            node = self.find_data_node(xpath)
+
+        if (node):
+            node.unlink()
+            dnode = self.find_data_node(xpath)
+            if (dnode is None):
+                #deleted node not found
+                return True
+            else:
+                print('Could not delete Node')
+                return False
+        else:
+            print("failed to find node, xpath: " + xpath)
+
+        return False
+
+    """
+    find_data_node_value():  find the value of a node from the data tree
+    input:    data_xpath of the data node
+    returns:  value string of the node
+    """
+    def find_data_node_value(self, data_xpath):
+        output = ""
+        try:
+            data_node = self.find_data_node(data_xpath)
+        except Exception as e:
+            print("find_data_node_value(): Failed to find data node from xpath: {}".format(data_xpath))
+            self.fail(e)
+        else:
+            if (data_node is not None):
+                subtype = data_node.subtype()
+                if (subtype is not None):
+                    value = subtype.value_str()
+                    return value
+            return output
+
+    """
+    set the value of a node in the data tree
+    input:    xpath of the data node
+    returns:  Exception if failed
+    """
+    def set_data_node_value(self, data_xpath, value):
+        try:
+            self.root.new_path(self.ctx, data_xpath, str(value), ly.LYD_ANYDATA_STRING, ly.LYD_PATH_OPT_UPDATE)
+        except Exception as e:
+            print("set data node value failed for xpath: " + str(data_xpath))
+            self.fail(e)
+
+    """
+    find_data_nodes(): find the set of nodes for the xpath
+    input:    xpath of the data node
+    returns:  list of xpath of the dataset
+    """
+    def find_data_nodes(self, data_xpath):
+        list = []
+        node = self.root.child()
+        try:
+            node_set = node.find_path(data_xpath);
+        except Exception as e:
+            self.fail(e)
+        else:
+            if node_set is None:
+                raise Exception('data node not found')
```

Comment:
should we define our own expection, like SonicYangException?

Reply:
sound good to me, I will add a SonicYangException class.

Reply:
[sonic_yang.py]: Categories public and private functions.

    Changes:

    1.) Categories public and private functions.
    2.) Minor spaces related changes.
    3.) Defining class SonicYangException and using it in all public functions.
    4.) Changing Class names to PascalCase.
    5.) Keeping functions, which deals with Libyang, in lower_underscore_case, rest functions names are in camelCase.